packages/diagrams/backend/sirius-components-diagrams-layout/src/main/java/org/eclipse/sirius/components/diagrams/layout/experimental/DiagramLayoutEngine.java

```diff
@@ -43,13 +57,131 @@ public DiagramLayoutEngine(IDiagramLayoutConfigurationProvider diagramLayoutConf
+
+        // 1. It should be enough for our actual contents with the border and padding accounted for
+        Size contentsSize = new Size(childrenFootprint.bottomRight().x() + internalOffsets.right(), childrenFootprint.bottomRight().y() + internalOffsets.bottom());
+        // 2. It can not be less than the minimum requested size. The diagram itself has no minimum size.
+        Size minSize = Optional.ofNullable(input.getNodeLayoutConfiguration(containerId)).map(NodeLayoutConfiguration::getMinimumSize).orElse(new Size(Double.MAX_VALUE, Double.MAX_VALUE));
+        // 3. If we had a previously set size, it must also be considered
+        Size previousSize = input.getPreviousBounds(containerId).map(Rectangle::size).orElse(new Size(0, 0));
+        // 4. The end-user may have asked for a new size explicitly
+        Size requestedSize = input.getOptionalResizeEvent(containerId).map(ResizeEvent::newSize).map(s -> new Size(s.getWidth(), s.getHeight())).orElse(previousSize);
+        double width = Math.max(contentsSize.width(), Math.max(minSize.width(), requestedSize.width()));
+        double height = Math.max(contentsSize.height(), Math.max(minSize.height(), requestedSize.height()));
+        nodeBounds.put(containerId, new Rectangle(0, 0, width, height));
+    }
+
+    /**
+     * Assuming all the children have their proper size, arrange them (only changing their positions) to their final
+     * position (relative to their parent).
+     */
+    private void arrangeChildren(String parentElementId, List<String> children, DiagramLayoutInput input, Map<String, Rectangle> layout) {
+        Canvas canvas = new Canvas();
+
+        // First, place the node(s) which has been directly interacted with by the end-user if there are any.
+        for (String childId : children) {
+            Optional<MoveEvent> optionalMoveEvent = input.getOptionalMoveEvent(childId);
+            if (optionalMoveEvent.isPresent()) {
+                Position newPosition = new Position(optionalMoveEvent.get().newPosition().getX(), optionalMoveEvent.get().newPosition().getY());
+                canvas.setBounds(childId, layout.get(childId).moveTo(newPosition));
+            }
+            // A resize can also change the position if it moves the top-left corner.
+            Optional<ResizeEvent> optionalResizeEvent = input.getOptionalResizeEvent(childId);
+            if (optionalResizeEvent.isPresent()) {
+                ResizeEvent resizeEvent = optionalResizeEvent.get();
+                input.getPreviousBounds(childId).ifPresent(bounds -> {
+                    double dx = -resizeEvent.positionDelta().getX();
+                    double dy = -resizeEvent.positionDelta().getY();
+                    Position newPosition = bounds.topLeft().translate(dx, dy);
+                    canvas.setBounds(childId, layout.get(childId).moveTo(newPosition));
+                });
+            }
+        }
```

Comment:
When a child is moved upper left, it can increase the size of the parent and also moving all children the amount of the top left offset